rust/util/mod.rs

```diff
@@ -32,6 +32,17 @@ macro_rules! enum_getter {
     };
 }
 
+#[macro_export]
+macro_rules! enum_wrapper {
+    {$enum_name:ident $($classname:ty => $enum_value:ident),* $(,)*} => {
+        $(impl From<$classname> for $enum_name {
+            fn from(x: $classname) -> Self {
+                Self::$enum_value(x)
+            }
+        })*
+    };
+}
+
```

Comment:
This macro allows us to quickly generate the `From` trait implementations for enums.

Reply:
Ah, is this a macro that lets us easily implement `From` to, in a way, "downcast" an enum to one of its members (or the other way, "upcasting" an enum variant to the parent enum)? I've been looking for one of these.

If so, is there not a crate already out there that provides such functionality?

Reply:
Not that I've found. There's `enum-as-inner` that does the same thing as the `enum_getter` macro. Either way, I hesitate adding more dependencies for something this trivial.